### PR TITLE
Widget block show author

### LINKED ISSUE #956: Block Editor: Widget author needed
Screenshot 2020-01-07 at 17 59 09

To recreate the issue, install and activate LiveMesh SiteOrigin Widgets. Go to Plugins > SiteOrigin Widgets and activate all of the LiveMesh widgets.

At Plugins > SiteOrigin Widgets, it's no problem, we list the author:

SiteOrigin_Widgets_‹_SiteOrigin_—_WordPress

We need something similar for the Widgets Bundle Block.

### PR DESCRIPTION
**Pull request by braamgenis**
Resolve #956 
Append third party widget author names to distinguish between SiteOrigin and third party widgets.

### CODE REVIEW

File: compat/block-editor/widget-block.php [hunk 1]
```diff
@@ -8,40 +8,70 @@ class SiteOrigin_Widgets_Bundle_Widget_Block {
...
+					$third_party_widgets[] = array(
+						'name' => $name,
+						'class' => $class,
+					);
+				}
 			}
 		}
-		
+		// Sort the list of widgets so SiteOrigin widgets are at the top and then third party widgets.
+		sort($so_widgets);
```
Comment (Misplon) [inline compat/block-editor/widget-block.php]: Minor formatting change here if possible. `sort($so_widgets);` to `sort( $so_widgets );`. Thanks.

**Comment by Misplon** [comment]
At Plugins > SiteOrigin Widgets we aren't ordering our widgets first then other authors. Are you sure we want to do this for the SO Widget Block? An alternative would be to list the widgets alphabetically but only list SiteOrigin as the author of SO widgets when non-SiteOrigin widgets are detected.

**Comment by braamgenis** [comment]
@Misplon I had it listing them alphabetically and I think it was pretty messy, with or without displaying _by SiteOrigin_ alongside SO widgets. Sorting the widgets like this seemed neater to me. I've attached screenshots to compare.

Alphabetical:
Screen Shot 2020-01-15 at 12 55 35

By Author:
Screen Shot 2020-01-15 at 12 56 46

**Comment by Misplon** [comment]
Given the different format to Plugins > SiteOrigin Widgets (very limited space) and that we can change this later should we come to a different opinion yeah, this is good to go.

**Comment by Misplon** [comment]
Just awaiting a quick review from @gregpriday.

**Pull request merged**